app/src/main/java/protect/card_locker/MainActivity.java

```diff
@@ -532,18 +530,36 @@ public boolean onOptionsItemSelected(MenuItem inputItem)
                 }
             }
 
-            builder.setSingleChoiceItems(R.array.sort_types_array, currentIndex.get(), (dialog, which) -> currentIndex.set(which));
-            builder.setPositiveButton(R.string.sort, (dialog, which) -> {
-                setSort(loyaltyCardOrders.get(currentIndex.get()), DBHelper.LoyaltyCardOrderDirection.Ascending);
+            AlertDialog.Builder builder = new AlertDialog.Builder(MainActivity.this);
+            builder.setTitle(R.string.sort_by);
 
-                dialog.dismiss();
+            final View customLayout = getLayoutInflater().inflate(R.layout.sorting_option, null);
+            builder.setView(customLayout);
+
+            CheckBox ch = (CheckBox) customLayout.findViewById(R.id.checkBox_reverse);
+            ch.setChecked(getSharedPreferences(getString(R.string.sharedpreference_reverse_state), Context.MODE_PRIVATE).getBoolean("ch", true));
+            ch.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
+                @Override
+                public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
+                    getSharedPreferences(getString(R.string.sharedpreference_reverse_state), Context.MODE_PRIVATE).edit().putBoolean("ch", isChecked).apply();
```

Comment:
There is no need for this, the `setSort` function already saves sharedpreference_sort_direction.

Reply:
That is used to save the state of the checkbox. If I don't use this then the checkbox is won't stay checked

Reply:
We don't want to save the state of the checkbox though. When creating the dialog, you should check what the current sort direction is. If it is Descending, check the checkbox, if it is Ascending do not. That we the checkbox can also never end up out-of-sync with reality.